pandas/tests/indexes/datetimes/test_construction.py

```diff
@@ -33,16 +33,15 @@ def test_construction_with_alt(self):
         tm.assert_index_equal(i, i2)
         assert i.tz.zone == 'US/Eastern'
 
-        i2 = DatetimeIndex(i.tz_localize(None).asi8, tz=i.dtype.tz)
+        i2 = DatetimeIndex(i.asi8, tz=i.dtype.tz)
         tm.assert_index_equal(i, i2)
         assert i.tz.zone == 'US/Eastern'
 
-        i2 = DatetimeIndex(i.tz_localize(None).asi8, dtype=i.dtype)
```

Comment:
i actually like this construction, can you add a test that does both? (e.g. both uses and does not localize), maybe split it out and parameterize